core/src/main/java/org/everit/json/schema/internal/RelativeJsonPointerFormatValidator.java

```diff
@@ -0,0 +1,102 @@
+package org.everit.json.schema.internal;
+
+import java.util.Optional;
+
+import org.everit.json.schema.FormatValidator;
+import org.json.JSONPointer;
+
+public class RelativeJsonPointerFormatValidator implements FormatValidator {
+
+    private static class ParseException extends Exception {
+
+        public ParseException(String input) {
+            super(String.format("[%s] is not a valid relative JSON Pointer", input));
+        }
+    }
+
+    private static final class Parser {
+
+        private static boolean isDigit(char c) {
+            return '0' <= c && c <= '9';
+        }
+
+        private String input;
+
+        private int pos = 0;
+
+        public Parser(String input) {
+            this.input = input;
+        }
+
+        public void parse() throws ParseException {
+            parseUpwardsStepCount();
+            parseJsonPointer();
+            parseTrailingHashmark();
+        }
+
+        private void parseTrailingHashmark() throws ParseException {
+            if (pos == input.length()) {
+                return;
+            }
+            if (pos == input.length() - 1 && input.charAt(pos) == '#') {
+                return;
+            }
+            fail();
+        }
+
+        private char curr() {
+            return input.charAt(pos++);
+        }
+
+        private void parseUpwardsStepCount() throws ParseException {
+            if (!isDigit(curr())) {
+                fail();
+            }
+            if (pos == input.length()) {
+                return;
+            }
+            for (char current = curr(); isDigit(current) && pos < input.length() - 1; current = curr())
+                ;
+            pos--;
```

Comment:
Does it work for values that consists of only positive-integer and it has more than one digits e.g. '123'? In that case I think because of the step back the other parsing (parseJsonPointer) will try to process the ending digit.

This could be prevented by something like:
` if (!(pos == input.length() - 1)) {
pos--;
}`